### PR TITLE
circleci: basic macos build [ECR-443]

### PR DESCRIPTION
**Pull request by ozkriff**
This PR adds some basic macOS CI testing.

### CODE REVIEW

**Comment by aleksuss** [comment]
I guess you should add env variables for precompiled libraries `libsodium` and `librocksdb`.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+version: 2
+
+jobs:
+  build_and_test:
+    macos:
+      xcode: "9.0"
```
Comment (aleksuss) [inline .circleci/config.yml]:
```
environment:
  - ROCKSDB_LIB_DIR: /usr/local/lib
  - SNAPPY_LIB_DIR: /usr/local/lib
```

**Comment by vitvakatu** [comment]
Shouldn't we do the same thing as in Travis: https://github.com/exonum/exonum/blob/f81d4535a4173651e8cbb28d7b00cfdbd2db0b45/.travis.yml#L104

**Comment by matklad** [comment]
>Shouldn't we do the same thing as in Travis: 

I think we can drop `--all` from `cargo test` on travis: `--all` is implied for virtual manifests. OTOH, `cargo fmt` and `cargo clippy` seem to need an explicit `--all` :(

**Comment by vitvakatu** [comment]
@matklad actually I meant running examples as well
cc @ozkriff

**Comment by DarkEld3r** [comment]
@vitvakatu Running examples would be nice, but we have pretty harsh limits on CircleCI, so I suppose it can be dropped (for now).

**Pull request merged**